src/lib/badge/_badge-theme.scss

```diff
@@ -126,6 +128,20 @@ $mat-badge-large-size: $mat-badge-default-size + 6;
     }
   }
 
+  .mat-badge-disabled {
+    .mat-badge-content {
+      // The disabled color usually has some kind of opacity, but because the badge is overlayed
+      // on top of something else, it won't look good if it's opaque. We convert it into a solid
+      // color by taking the opacity from the rgba value and using the value to determine the
+      // percentage of the background to put into foreground when mixing the colors together.
+      $app-background: mat-color($background, 'background');
+      $badge-color: mat-color($foreground, disabled-button);
+      $badge-opacity: opacity($badge-color);
+      background: mix($app-background, rgba($badge-color, 1), (1 - $badge-opacity) * 100%);
```

Comment:
Would it make sense to do alpha-blending here?
https://gist.github.com/jelbourn/f890a79a48cb21a5246a
https://en.wikipedia.org/wiki/Alpha_compositing#Alpha_blending

(I'm just looking for an excuse to use that sass function I wrote ages ago)

Reply:
Is that basically what the expression here is doing?

Reply:
I'm not sure since I don't know how the sass `mix` function works, but it's probably either close enough or the same thing